### PR TITLE
Add int2 support

### PR DESCRIPTION
**Pull request by superbobry**
As proposed in RFC: int2 in StableHLO (#2403), this PR adds support for these types to StableHLO.

### CODE REVIEW

File: stablehlo/dialect/VhloTypes.td [hunk 1]
```diff
@@ -119,6 +119,9 @@ def VHLO_FunctionV1 : VHLO_TypeDef<"FunctionV1", "func_v1", "0.9.0", "current">
 // dynamism RFC.
 def VHLO_IndexV1 : VHLO_TypeDef<"IndexV1", "index_v1", "0.9.0", "current">;
 
+// Corresponds to the 'si2' FloatType from the StableHLO spec.
```
Comment (sdasgup3) [inline stablehlo/dialect/VhloTypes.td:122]: s/FloatType/IntegerType/
here and below

File: stablehlo/dialect/Base.td [hunk 2]
```diff
@@ -38,8 +38,8 @@ def HLO_Pred : TypeAlias<I1, "pred (AKA boolean or 1-bit integer)">;
 
 // TODO(hinsu): Use signed integers instead of signless integer which is being
 // used for legacy reasons.
-def HLO_SInt : SignlessIntOfWidths<[4, 8, 16, 32, 64]>;
-def HLO_UInt : UnsignedIntOfWidths<[4, 8, 16, 32, 64]>;
+def HLO_SInt : SignlessIntOfWidths<[2, 4, 8, 16, 32, 64]>;
+def HLO_UInt : UnsignedIntOfWidths<[2, 4, 8, 16, 32, 64]>;
```
Comment (GleasonK) [inline stablehlo/dialect/Base.td:42]: Can you also add 2bit to:

```
def HLO_QuantizedSignedInt : StableHLO_UniformQuantizedSignedIntOfWidths<[4, 8, 16, 32]>;
def HLO_QuantizedUnsignedInt : StableHLO_UniformQuantizedUnsignedIntOfWidths<[4, 8, 16, 32]>;
```

**Review by sdasgup3 — APPROVED** [review]

File: stablehlo/dialect/Base.td (hunk 2 above)
Comment (superbobry) [reply stablehlo/dialect/Base.td:42]: Done.

File: stablehlo/dialect/VhloTypes.td (hunk 1 above)
Comment (superbobry) [reply stablehlo/dialect/VhloTypes.td]: Done. Good catch!

**Comment by superbobry** [comment]
@GleasonK all tests pass locally, so I think this is ready for review.

**Review by GleasonK — APPROVED** [review]

**Comment by GleasonK** [comment]
Will merge once CI passes! Looks great!

**Pull request merged**